test/app.js

```diff
@@ -19,33 +19,50 @@ const assert = require('assert');
 const backup = require('../app.js').backup;
 const fs = require('fs');
 const nock = require('nock');
+const util = require('util');
+const backupPromise = util.promisify(backup);
 
 const goodUrl = 'http://localhost:5984/db';
 // The real validateArgs function of app.js isn't
 // exported - so we call the exported backup method
 // instead. We don't get as far as a real backup when
 // testing error cases. For success cases we nock the
 // goodUrl and
-const validateArgs = function(url, opts, callback) {
+const validateArgs = async function(url, opts, callback) {
```

Comment:
I don't think we should be using a callback in an async function, even if it is just a way of passing the error around (the naming `callback` is confusing if nothing else because of the history here).

I think we can simplify this by taking advantage of `assert.rejects`: a proposal in https://github.com/IBM/couchbackup/commit/5f8b4e3d8ce517024d6aa299d1b98aab35a805ca